source/views/Extension/Views/ImportWallet/Steps/Step2.jsx

```diff
@@ -17,11 +24,26 @@ const Step2 = ({ userPemFile }) => {
   const [walletName, setWalletName] = useState(""); // add deafault wallet name, for example the next wallet not used number
   const [currentEmoji, setCurrentEmoji] = useState("😎"); // add default emoji, not used in other wallets
   const [openEmojiSelector, setOpenEmojiSelector] = useState(false);
+  const [importedPrincipal, setImportedPrincipal] = useState();
+  const [detailsModal, setDetailsModal] = useState(false);
 
   const { navigator } = useRouter();
 
   const classes = useStyles();
 
+  useEffect(() => {
+    let fileReader = new FileReader();
+    fileReader.readAsText(userPemFile);
+    fileReader.onloadend = () => {
+      const content = fileReader.result;
+
+      asyncSendMessage({
+        type: HANDLER_TYPES.GET_PRINCIPAL_FROM_PEM,
+        params: { pem: content },
+      }).then(setImportedPrincipal);
+    };
+  }, [userPemFile]);
```

Comment:
shouldn't this be wrapped under a redux action?

Reply:
mm We're not using redux for the import pem